src/nvector/serial/nvector_serial.c

```diff
@@ -171,29 +164,24 @@ N_Vector N_VNewEmpty_Serial(sunindextype length, SUNContext sunctx)
 
 N_Vector N_VNew_Serial(sunindextype length, SUNContext sunctx)
 {
+  SUNFunctionBegin(sunctx);
   N_Vector v;
   sunrealtype* data;
 
+  SUNAssertNull(length >= 0, SUN_ERR_ARG_OUTOFRANGE);
+
   v = NULL;
   v = N_VNewEmpty_Serial(length, sunctx);
-  if (v == NULL) { return (NULL); }
+  SUNCheckLastErrNull();
 
   /* Create data */
-  if (length > 0)
-  {
-    /* Allocate memory */
-    data = NULL;
-    data = (sunrealtype*)malloc(length * sizeof(sunrealtype));
-    if (data == NULL)
-    {
-      N_VDestroy_Serial(v);
-      return (NULL);
-    }
+  data = NULL;
+  data = (sunrealtype*)malloc(length * sizeof(sunrealtype));
+  SUNAssertNull(data, SUN_ERR_MALLOC_FAIL);
```

Comment:
The assert above allows `length = 0`, so should this `malloc` and corresponding attach operations still be protected inside `if (length > 0)`?